### PR TITLE
Julia 1.7.0, plus Arm version

### PR DESCRIPTION
**Pull request by pthariensflame**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

### CODE REVIEW

**Comment by singularitti** [comment]
Hi, I do not want to block this PR, just have one question. The current Julia still does not have some packages working on the Arm architecture. So I still want to install the Intel version Julia, on my M1 macs. Is there an option that I can choose from?

**Comment by bevanjkay** [comment]
@singularitti I don't believe there is anyway to force `homebrew` to download the intel binaries. cc: @carlocab

**Comment by carlocab** [comment]
If you install Homebrew in `/usr/local` and run it under Rosetta it should install the Intel version.

File: Casks/julia.rb [hunk 1]
```diff
@@ -1,15 +1,23 @@
 cask "julia" do
-  version "1.6.4"
-  sha256 "a00598f04832b8f5d28039942280feadf40e8a14d7f6264c754ec70627507ace"
+  arch = Hardware::CPU.intel? ? "mac64" : "macaarch64"
+  arch_folder = Hardware::CPU.intel? ? "x64" : "aarch64"
```
Comment (carlocab) [inline Casks/julia.rb:3]:
```suggestion
  arch, arch_folder = Hardware::CPU.intel? ? ["mac64", "x64"] : ["macaarch64", "aarch64"]
```
You can probably even combine this with the `if ... else ... end` block below, though that would be a bit nonstandard.

**Pull request merged**